pkg/api/security/security.go

```diff
@@ -117,10 +117,20 @@ func GetAuthTokenFilepath() string {
 // FetchAuthToken gets the authentication token from the auth token file & creates one if it doesn't exist
 // Requires that the config has been set up before calling
 func FetchAuthToken() (string, error) {
+	return fetchAuthToken(false)
+}
+
+// CreateOrFetchToken gets the authentication token from the auth token file & creates one if it doesn't exist
+// Requires that the config has been set up before calling
+func CreateOrFetchToken() (string, error) {
```

Comment:
I don't think this method called `CreateOrFetchToken` is ever used to fetch the token, we could consider not naming it this way? Has it been done this way to have the same behavior between the Infra Agent and the DCA?

Reply:
AFAIK the `auth_token` file is kept when the agent stops (not 100% sure). That was one of the main reason to call it CreateOrFetch and the other reason for this implementation was to be able to come up with a harmonized solution between the agent and the DCA.

Reply:
Got it! In all cases that was a nit.